### ISSUE TITLE
KeyError: 'return' when using SALTAPI_EAUTH=auto

### ISSUE THREAD

**Issue by arthurlogilab** [saltstack/pepper#19]
```
$ pepper '*' test.ping
Traceback (most recent call last):
  File "/home/arthur/.virtualenvs/pepper/bin/pepper", line 265, in <module>
    exit_code, results = main()
  File "/home/arthur/.virtualenvs/pepper/bin/pepper", line 235, in main
    nodesJid = nodesJidRet['return'][0]['jid']
KeyError: 'return'
```

switching SALTAPI_EAUTH=pam in .pepperrc gets it to work.

Would be nice to catch this traceback.

**Comment by whiteinge** [comment]
Yeah, that's no good. We're obviously not checking for certain types of errors there. Can you paste the actual return from salt-api that you're getting?

**Comment by arthurlogilab** [comment]
The -H doesn't work to get debug this HTTP (should I open another issue ? ). For some reason the logger.debug in libpepper doesn't print out on the console here (using -v -H). Added some prints and got a simple : 

```
Error with request HTTP Error 401: Unauthorized
```

salt-api side : 
```
[INFO    ] 127.0.0.1 - - [30/Jan/2015:10:55:16] "POST /login HTTP/1.1" 401 742 "" "Python-urllib/2.7"
[INFO    ] 127.0.0.1 - - [30/Jan/2015:10:55:16] "POST / HTTP/1.1" 401 735 "" "Python-urllib/2.7"
```

**Comment by whiteinge** [comment]
Thanks for filing this! I didn't realize the logging had been broken. Fixed in f623e84. A few error-handling enhancements have been made in 4cd1661 so you should see the correct 401 error now.

**Comment by whiteinge** [comment]
Please reopen if this is still an issue.
